docs/src/architecture/data-model.md

```diff
@@ -0,0 +1,152 @@
+
+# thin-edge Data Model
+
+The **data model** identifies all data send or received from/to **thin-edge** and it's components, to interact with those.
+For all data it defines format and explains behaviour.
+
+## Telemetry Data
+
+**Telemetry Data** are **measurements**, **events** and **alarms**. Each is defined by a set of data-elements and has it specific behaviour.
+
+### Measurements
+  * **measurements** carry values from physical **Sensors**[^1] or a device's **Domain Application**[^1];
+    e.g. voltage and current of an electricity meter, or current state of the manufacturing control process
+  * **elements of a measurement:**
+    - `type name`, a string that identifies the measurement uniquely in context of a device
+    - `samples`, carry a single value or set of values, all taken at a single point in time
+      - `values` can be a mix of numbers, strings or booleans
+        - optionally each value can have a `type name`
+      - `timestamp` (conform to ISO 8601), that indicates when values were sampled
+    - TODO: not yet supported: list of units per sample's values
+    - TODO: not yet supported: source
+  * **behaviour of a measurement:**
+    - thin-edge does not store any historical sampled values for measurements
+    - there is no initialization-value for measurements; i.E. a measurement is not visible on thin-edge before the 1st sample was sent to thin-edge
+
+### Events
+  * **events** are notifications that something happened on a device's environment or software system;
+    e.g. a sensor[^1] detected something like a door has been closed,
+    or a system notification that e.g. a user has started an ssh session
+  * **elements of an event:**
+    - `type name`, a string that identifies the event uniquely in context of a device
+    - `event text`, that carries a human readable event-message
+      - must be UTF-8 encoded
+    - `timestamp` (conform to ISO 8601), that indicates when the event has occured
```

Comment:
Is it worth mentioning that the `timestamp` can be automatically added by thin-edge.io when creating the measurement? So for simple cases, the user does not need to worry about the ISO8601 timestamp format when creating events/measurements/alarms (if they don't need to)?

Reply:
Done. See https://github.com/thin-edge/thin-edge.io/pull/1677/commits/10b6e223ac06eaf8d7dd3a47edd11df711bbc477